### DISCUSSION TITLE
Rate Limit exceeded and other errors on new Agent App

### DISCUSSION THREAD

**Discussion by manderson88** [iTwin/itwinjs-core#1708] [Q&A] [answered]
I am working on a new app based on the jumpstart agent application.  I have been getting a rate limit error when I try to run my newly created application.  What can I do to check how I am exceeding the limit for queries?

thanks,
marka

**Comment by calebmshafer** [comment]
hey @manderson88, can you check and see if your client id is upgraded in the developer portal? 

Updating the client can be done by going to https://developer.bentley.com/my-apps/ and clicking on the specific client. Once on the details page for the client, there should be an "Upgrade" button at the top of the page, if you click it and retry running your Agent the rate limit issues should go away.

Apologizes for the inconvenience and are working to alleviate this for all new and existing clients. Thanks!

**Comment by manderson88** [comment, marked as answer]
Solved it by creating a new application entry.  BTW never saw an upgrade button on the page.

**Reply by calebmshafer** [reply]
@manderson88, glad you were able to solve it.

@laurynasr any idea why the "Upgrade" button may have been missing?
